fix: Return error message when relationship is not accepted in Comple…

StergiosSozos opened the pull request:
### Description
The mentorship relation state (accepted or not) is now checked when someone is trying to complete a task from a relationship that is not accepted. This check is also of higher importance than the existence of the task. So if the relationship is not accepted the user gets a message that the relationship is not accepted.

Fixes #537

### Type of Change:
- Code

### How Has This Been Tested?
All the tests have the same result as before. No extra tests were added.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

StergiosSozos committed:
fix: Return error message when relationship is not accepted in Complete task #537

StergiosSozos committed:
Change codes to http to fix conflicts

isabelcosta commented on app/api/resources/mentorship_relation.py:
```diff
@@ -657,7 +657,14 @@ class UpdateTask(Resource):
-    @mentorship_relation_ns.response(HTTPStatus.BAD_REQUEST, "%s" % messages.TASK_WAS_ALREADY_ACHIEVED)
+    @mentorship_relation_ns.response(
+        HTTPStatus.BAD_REQUEST,
+        "%s\n%s"
```
Could you use f-strings, as per these [coding standards](https://github.com/anitab-org/mentorship-backend/wiki/Coding-Standards)

StergiosSozos replied:
I changed it to f-strings!

isabelcosta commented on app/api/dao/task.py:
```diff
@@ -136,7 +136,8 @@ def complete_task(user_id: int, mentorship_relation_id: int, task_id: int):
             return messages.USER_NOT_INVOLVED_IN_THIS_MENTOR_RELATION, HTTPStatus.UNAUTHORIZED
-
+        if relation.state != MentorshipRelationState.ACCEPTED:
+            return messages.UNACCEPTED_STATE_RELATION, HTTPStatus.BAD_REQUEST
```
add a blank line below please, just to separate visually each validation section

StergiosSozos replied:
I added the blank spaces!

isabelcosta reviewed (changes requested):
@StergiosSozos I requested some minor changes. this is almost done, thank you for this contribution 🎉

StergiosSozos committed:
Add blank spaces to separate visually each validation section

StergiosSozos committed:
Use f strings style

isabelcosta commented on app/api/dao/task.py:
```diff
@@ -131,13 +131,17 @@ def complete_task(user_id: int, mentorship_relation_id: int, task_id: int):
+        if relation.state != MentorshipRelationState.ACCEPTED:
+            return messages.UNACCEPTED_STATE_RELATION, HTTPStatus.BAD_REQUEST
         task = relation.tasks_list.find_task_by_id(task_id)
+
```
blank line was in line 143, and not 144

StergiosSozos replied:
Sorry for that, is it now correct?

StergiosSozos committed:
Add blank space to separate sections visually

isabelcosta reviewed (changes requested):
I just recalled now, can you create a simple test for it, showing that that is the first message that appears if the user attempts to create a task in a non accepted relation.
This would make this PR shine 🤩

StergiosSozos commented:
Sure, so a unit test  right? So that we can test that the message that we want to be diaplayed is the one i added here and it does appear.

isabelcosta commented:
> Sure, so a unit test right? So that we can test that the message that we want to be diaplayed is the one i added here and it does appear.

yes :)

StergiosSozos committed:
Test:Add test for complete_task when user's relation is pending
Create 5th user
Create relationship between 4th and 5th user that is not accepted
Add test case to check the message shown

StergiosSozos commented:
A unit test for my method is now added, in addition with an extra user on the databse.

devkapilbansal commented:
@StergiosSozos can you please update this PR

isabelcosta commented:
@StergiosSozos thank you for working on this project 🙌🏾 From what I see, only the merge conflicts are left to approve this. However, I will close this PR due to lack of response. If you wish to work on it again and the issue is still available feel free to comment there :)

mariejp commented:
Hi! What were the requested changes for this PR? I cannot seem to locate them anywhere.
